File: apps/frontend/src/app/foundation/support/support.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { ApiClient, ApiPage } from '@foundation/networking';
import { CreateSupportTicketRequest, SupportMessage, SupportTicket } from './support.models';

@Injectable({ providedIn: 'root' })
export class SupportService {
  private readonly api = inject(ApiClient);

  listTickets(page = 0, size = 20): Observable<ApiPage<SupportTicket>> {
    return this.api.get<ApiPage<SupportTicket>>('/support/tickets', {
      params: {
        page,
        size,
      },
    });
  }

  createTicket(payload: CreateSupportTicketRequest): Observable<SupportTicket> {
    return this.api.post<SupportTicket, CreateSupportTicketRequest>('/support/tickets', payload);
  }

  listMessages(ticketId: string): Observable<SupportMessage[]> {
    return this.api.get<SupportMessage[]>(
      `/support/tickets/${encodeURIComponent(ticketId)}/messages`,
    );
  }

  sendMessage(ticketId: string, body: string): Observable<SupportMessage> {
    return this.api.post<SupportMessage, { body: string }>(
      `/support/tickets/${encodeURIComponent(ticketId)}/messages`,
      { body },
    );
  }
}

File: apps/ios/SpotLink/Sources/SpotLink/Features/Support/SupportService.swift
import Foundation

// MARK: - Support Service

public final class SupportService: Sendable {
    private let apiClient: APIClientProtocol

    public init(apiClient: APIClientProtocol) {
        self.apiClient = apiClient
    }

    public func listTickets(page: Int = 0, size: Int = 20) async throws -> APIPage<SupportTicket> {
        try await apiClient.get("/support/tickets", query: ["page": String(page), "size": String(size)])
    }

    public func createTicket(_ request: CreateTicketRequest) async throws -> SupportTicket {
        try await apiClient.post("/support/tickets", body: request)
    }

    public func getMessages(ticketId: String) async throws -> [SupportMessage] {
        try await apiClient.get("/support/tickets/\(ticketId)/messages", query: nil)
    }

    public func addMessage(ticketId: String, body: String) async throws -> SupportMessage {
        try await apiClient.post("/support/tickets/\(ticketId)/messages", body: AddMessageRequest(body: body))
    }
}
